consul_kv: decodes kv values from Consul to utf-8

colin-nolan opened the pull request:
##### SUMMARY
<!--- Describe the change, including rationale and design decisions -->
Fixes #35893 by decoding values from Consul as utf-8.

##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - Bugfix Pull Request

##### COMPONENT NAME
<!--- Name of the module, plugin, module or task -->
consul_kv

##### ANSIBLE VERSION
<!--- Paste verbatim output from "ansible --version" between quotes below -->
```
ansible 2.4.2.0
  config file = None
  configured module search path = ['/root/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.6/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 3.6.2 (default, Sep 13 2017, 14:26:54) [GCC 4.9.2]
```


##### ADDITIONAL INFORMATION
<!---
Include additional information to help people understand the change here.
For bugs that don't have a linked bug report, a step-by-step reproduction
of the problem is helpful.
  -->
N/A

Commit:
Decodes kv values from Consul to utf-8.

abadger commented:
This looks like the right idea but you should use our helper functions for encoding and decoding text.  When you do that, the code will look like this:

``` python
from ansible.module_utils._text import to_text
[...]
try:
    changed = not existing or (existing and to_text(existing['Value'], errors='surrogate_or_strict') != value)
except UnicodeError:
    # Existing value was undecodable but all values we set are valid utf-8
    changed = True
```

colin-nolan commented:
@abadger thanks. Why does the `_text` module name suggest that it's (package) private? It does not look like somewhere I should be importing things from for general use!

abadger commented:
It's named such so that the API can be changed or the package moved to a
different directory without upsetting third parties too much,. (for modules
in the repo, I'll port the modules if such a thing happens,)

The API has proved to be suitably stable for enough releases now that I'm
ready to remove the leading underscore for that.  However we're also
planning to move a few modules into a new subdirectory,
ansible/module_utils/common, so I've been holding off until we're ready to
do that.

On Tue, Mar 20, 2018, 3:18 AM Colin Nolan <notifications@github.com> wrote:

> @abadger <https://github.com/abadger> thanks. Why does the _text module
> name suggest that it's (package) private? It does not look like somewhere I
> should be importing things from for general use!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ansible/ansible/pull/35894#issuecomment-374544809>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAMxWtz6n3uYYyZPWF9-eUynxw0aqEv1ks5tgNd9gaJpZM4R-Hjw>
> .
>

Commit:
Switches to using module utils to perform text decoding.

Commit:
Adds self to authors list to help community maintenance.

colin-nolan commented:
Updated to use decoding helper. Also added self to authors list to speed up acceptance of future community PRs (I'm already an author on `consul_acl`).
